### PR TITLE
ContextAction.isEnabled needs a context

### PR DESCRIPTION
**Pull request by Hixie**
...and lots of things that fall out from that

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 1]
```diff
@@ -320,6 +320,10 @@ class Scrollable extends StatefulWidget {
   /// the nearest enclosing [ScrollableState] in that [Axis] is returned, or
   /// null if there is none.
   ///
+  /// This finds the nearest _ancestor_ [Scrollable] if the `context`. This
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/widgets/scrollable.dart:323]: fixed

File: packages/flutter/lib/src/widgets/actions.dart [hunk 2]
```diff
@@ -1540,12 +1599,17 @@ class PrioritizedIntents extends Intent {
 
 /// An [Action] that iterates through a list of [Intent]s, invoking the first
 /// that is enabled.
-class PrioritizedAction extends Action<PrioritizedIntents> {
+///
+/// The [isEnabled] method must be called before [invoke]. Calling [isEnabled]
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/widgets/actions.dart:1603]: I don't disagree but this PR doesn't affect any of that, I'm just documenting reality here.

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 3]
```diff
@@ -359,6 +363,10 @@ class Scrollable extends StatefulWidget {
   /// target [Scrollable] is not the closest instance. When [axis] is provided,
   /// the nearest enclosing [ScrollableState] in that [Axis] is returned.
   ///
+  /// This finds the nearest _ancestor_ [Scrollable] if the `context`. This
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/widgets/scrollable.dart:366]: fixed

File: packages/flutter/lib/src/widgets/actions.dart [hunk 4]
```diff
@@ -598,20 +631,47 @@ class ActionDispatcher with Diagnosticable {
...
+  /// is used.
+  ///
+  /// The return value has two components. The first is a boolean indicating if
+  /// the action was enabled (as per [Actions.isEnabled]). If this is false, the
+  /// second return value is null. Otherwise, the second return value is the
+  /// object returned from [Action.invoke].
+  ///
+  /// Consider using [invokeAction] if the enabled state of the action is not in
+  /// question; this avoids calling [Action.isEnabled] redundantly.
+  (bool, Object?) invokeActionIfEnabled(
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/widgets/actions.dart:665]: yeah i struggled with how to implement this for ages until i realised the answer was just to return... both pieces of information :-)

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by Hixie** [comment]
Web failures are caused by https://github.com/dart-lang/sdk/issues/52593.

**Pull request merged**